specification/dsdl/grammar.tex

```diff
@@ -0,0 +1,253 @@
+    Symbol                             & Precedence & Description \\
+    \texttt{\textbf{+}}                         & 3 & Unary plus \\
+    \texttt{\textbf{-}} (hyphen-minus)          & 3 & Unary minus \\
+    \texttt{\textbf{!}}                         & 8 & Logical not \\
+\end{UAVCANSimpleTable}
+
+\begin{UAVCANSimpleTable}{Binary operators}{|l l X|}
+    Symbol                                          & Precedence & Description \\
+    \texttt{\textbf{.}} (full stop)                          & 1 & Attribute reference
+                                                                   (parent object on the left side,
+                                                                   attribute identifier on the right side) \\
+
+    \texttt{\textbf{**}}                                     & 2 & Exponentiation
+                                                                   (base on the left side, power on the right side) \\
+
+    \texttt{\textbf{*}}                                      & 4 & Multiplication \\
+    \texttt{\textbf{/}}                                      & 4 & Division \\
+    \texttt{\textbf{\%}}                                     & 4 & Modulo \\
+
+    \texttt{\textbf{+}}                                      & 5 & Addition \\
+    \texttt{\textbf{-}} (hyphen-minus)                       & 5 & Subtraction \\
+
+    \texttt{\textbf{|}} (vertical line)                      & 6 & Bitwise or \\
+    \texttt{\textbf{\textasciicircum{}}} (circumflex accent) & 6 & Bitwise xor \\
+    \texttt{\textbf{\&}}                                     & 6 & Bitwise and \\
+
+    \texttt{\textbf{==}} (dual equals sign)                  & 7 & Equality \\
+    \texttt{\textbf{!=}}                                     & 7 & Inequality \\
+    \texttt{\textbf{<=}}                                     & 7 & Less or equal \\
+    \texttt{\textbf{>=}}                                     & 7 & Greater or equal \\
+    \texttt{\textbf{<}}                                      & 7 & Less \\
+    \texttt{\textbf{>}}                                      & 7 & Greater \\
+
+    \texttt{\textbf{||}} (dual vertical line)                & 9 & Logical or \\
+    \texttt{\textbf{\&\&}}                                   & 9 & Logical and \\
+\end{UAVCANSimpleTable}
+
+\subsection{Literals}
+
+Upon its evaluation, a literal yields an object of a particular type depending on the syntax of the literal,
+as specified in this section.
+
+\subsubsection{Boolean literals}
+
+A boolean literal is denoted by the keyword ``\verb|true|'' or ``\verb|false|''
+represented by an instance of primitive type ``\verb|bool|'' (section \ref{sec:dsdl_primitive_types})
+with an appropriate value.
+
+\subsubsection{Numeric literals}
+
+Integer and real literals are represented as instances of type ``\verb|rational|'' (section \ref{sec:dsdl_rational}).
+
+The digit separator character ``\verb|_|'' (underscore) does not affect the interpretation of numeric literals.
```

Comment:
underscore to separate digits? Don't you mean comma?

Reply:
`123_456_789` is a literal number which matches `literal_integer_decimal` and is interpreted as 123456789. They use underscore in Python like this, too. In C++ they use the single quote character `'` instead, which might read better but it's incompatible with our usage of single quotes for string literals. Okay, it's compatible, but the grammar gets kind of weird.

`123,456,789` is an expression which matches `expression_list = (expression (_? "," _? expression)*)?`, not a numeric literal.